src/mono/mono/mini/mini-arm64.c

```diff
@@ -2081,10 +2081,24 @@ mono_arch_set_native_call_context_ret (CallContext *ccontext, gpointer frame, Mo
 			storage = alloca (temp_size);
 		else
 			storage = arg_get_storage (ccontext, ainfo);
+
+#ifdef MONO_ARCH_HAVE_SWIFTCALL
+		int swift_error_preserved_val = 0;
+		if (ccontext->gregs [PARAM_REGS + 2]) {
+			swift_error_preserved_val = ccontext->gregs [PARAM_REGS + 2];
+		}
+#endif
+
 		memset (ccontext, 0, sizeof (CallContext)); // FIXME
```

Comment:
The whole `swift_error_preserved` thing is to circumvent this memset. Do you have any better ideas how to go about it?

Reply:
Unfortunately I don't remember what is the point of this memset. I would get rid of it instead.

Reply:
Perhaps update the memset num to be `sizeof (CallContext) - CTX_REGS` when `MONO_ARCH_HAVE_SWIFTCALL`?